### PR TITLE
CLOUDSTACK-8605: KVM: Config Drive and getVmIp support

### PR DESCRIPTION
**Pull request by kishankavala**
- CLOUDSTACK-8324 added support for External DHCP/DNS support along with ConfigDrive for XenServer
- FS link: https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=53740797
- This PR adds KVM support for config drive and getVmIpAddress assigned by external DHCP
- genisoimage package dependency is added for creating config drive ISO
- libguestfs is used to get IPAddress from guest VM. Linux: From dhcp leases file. Windows: From registry

### CODE REVIEW

**Comment by kishankavala** [comment]
/var/lib/libvirt/images is used for local storage also.  
com.cloud.hypervisor.kvm.resource.LibvirtComputingResource#configure
      _localStoragePath = (String)params.get("local.storage.path");
        if (_localStoragePath == null) {
            _localStoragePath = "/var/lib/libvirt/images/";
        }

I'll update the patch to use _localStoragePath  instead of hard-coded /var/lib/libvirt/images

File: packaging/centos63/cloud.spec [hunk 1]
```diff
@@ -129,6 +129,8 @@ Requires: perl
 Requires: libvirt-python
 Requires: qemu-img
 Requires: qemu-kvm
+Requires: genisoimage
```
Comment (bhaisaab) [inline packaging/centos63/cloud.spec]: you'll need to add this to other cloud.spec files in packages/{centos7,fedora20,fedora21}

**Comment by kishankavala** [comment]
@wido @bhaisaab Made the suggested changes

**Comment by wido** [comment]
Code-wise I'm not to happy. There are all kinds of assumptions about paths. mkisofs for example always being there in /usr/bin.

Using /tmp for temporary directories, who says that /tmp is big enough on every system? Always using /var/lib/libvirt to place the ISO? Why not fetch the local storage pool and figure out what the path is.

It's not guaranteerd that it will be /var/lib/libvirt/images on all systems.

Imho there are to many assumptions in the code which makes it fragile.

**Comment by remibergsma** [comment]
@kishankavala Any update on this?

**Comment by kishankavala** [comment]
Updated code to use local.storage.path config instead of hard-coded /var/lib/libvirt/images/. Local Storage pool is also created using same config. Removed /usr/bin path for mkisofs.
Other issue was regarding using /tmp for temporary directories. Should I change this also to use local storage path?

**Comment by kishankavala** [comment]
@remibergsma I've update the PR. There is still one issue open regarding the usage of /tmp

**Comment by remibergsma** [comment]
@kishankavala Thanks for the update! When you're ready, also be sure to ping @wido so he can have another look.

**Comment by bhaisaab** [comment]
LGTM.

**Comment by bhaisaab** [comment]
maybe rebase and fix any issues, to get Travis green before merging this once @wido can review this.

**Comment by remibergsma** [comment]
@kishankavala Please remove the 4 merge commits, thanks.

**Comment by remibergsma** [comment]
@kishankavala There are 5 merge commits now... please remove them.

**Pull request closed without merging**